pkg/kv/kvserver/logstore/sideload_disk.go

```diff
@@ -97,13 +101,26 @@ func (ss *DiskSideloadStorage) Put(
 		}
 		// Ensure that ss.dir exists. The filename() is placed directly in ss.dir,
 		// so the next loop iteration should succeed.
-		if err := ss.eng.MkdirAll(ss.dir, os.ModePerm); err != nil {
+		if err := mkdirAllAndSyncParents(ss.eng, ss.path[0], ss.path[1:], os.ModePerm); err != nil {
 			return err
 		}
 		continue
 	}
 }
 
+// Sync implements SideloadStorage.
+func (ss *DiskSideloadStorage) Sync() error {
+	dir, err := ss.eng.OpenDir(ss.dir)
+	if err != nil {
+		return err
+	}
```

Comment:
There are some cases when `ss.dir` is not yet, or already not existing. This code must account for that.